### DISCUSSION TITLE
Will a `quarto convert` round-trip from .qmd -> .ipynb -> .qmd generally yield an identical file?

### DISCUSSION THREAD

**Discussion by hugetim** [quarto-dev/quarto-cli#11228] [Q&A] [answered]
### Description

I expected to find tests of this but wasn't able to on a quick search. General round-trip consistency (preserving line breaks and white space, for example) would ease collaboration between people who prefer working with different formats. For example, see: https://github.com/posit-dev/positron/issues/1533#issuecomment-2212485898

**Comment by cscheid** [comment, marked as answer]
We'd like for that to be the case. But, frankly speaking, it's easier said than done. In general, the two formats are simply not equivalent.

We might be able to offer something (admittedly significantly) weaker, like `convert(convert(convert(x))) = convert(x)`, but `convert(convert(x)) = x` is pretty hard because that requires pinning down the behavior for _every_ document `x`, and there's a large degree of freedom in how that works.

**Reply by hugetim** [reply]
The weaker thing would be helpful, I think. I guess you essentially get a standardized format by doing `convert(convert(x))`. Maybe there could even be a shortcut in the VSCode extension or elsewhere to 'Apply Quarto Format'.

**Comment by hugetim** [comment]
For reference, the [Jupytext docs](https://jupytext.readthedocs.io/en/latest/formats-markdown.html#quarto) currently state:

> Note that the round trip of .ipynb to .qmd to .ipynb has the effect of concatenating consecutive Markdown cells and turning raw cells into Markdown cells (since .qmd files represent all content as either Markdown or code cells).

Jupytext also appears to have some round-trip tests:

- https://github.com/mwouts/jupytext/blob/b74f5ce78ed5b7daf770da2550ecd1b973cee99c/tests/external/round_trip/test_mirror_external.py#L17C1-L22C40
- https://github.com/mwouts/jupytext/tree/b74f5ce78ed5b7daf770da2550ecd1b973cee99c/tests/data/notebooks/outputs/ipynb_to_quarto (compare inputs [here](https://github.com/mwouts/jupytext/tree/b74f5ce78ed5b7daf770da2550ecd1b973cee99c/tests/data/notebooks/inputs/ipynb_py) and [here](https://github.com/mwouts/jupytext/tree/b74f5ce78ed5b7daf770da2550ecd1b973cee99c/tests/data/notebooks/inputs/ipynb_julia))
